### DISCUSSION TITLE
Gas Limit & Usage

### DISCUSSION THREAD

**Discussion by Moemen12** [Cyfrin/foundry-full-course-cu#2067] [Show and tell]
When I was watching the tutorial and sending 0.05 ETH to my metamask wallet , so i noticed that it used 100% of gaz limit but with the instructor of course it only uses 35% so why ?

![Screenshot 2024-07-21 013922](https://github.com/user-attachments/assets/10d21b61-76fd-4bc3-8584-7739f96ab26a)

**Comment by EngrPips** [comment]
It seems like you figured it out, right?

**Reply by Moemen12** [reply]
Yes, so why for 0.05 Eth , it withdrew 100% of Gaz

**Reply by Shiva-Sai-ssb** [reply]
Gas costs vary for different transactions on the blockchain. The simplest transaction is sending ETH from one account to another. The gas cost for this is fixed at 21,000 gas, regardless of whether you send 0.05 ETH or 50 ETH. Typically, MetaMask and other wallets sets the default gas limit to 21,000 gas for ETH transfers, which is usually sufficient to execute the transaction successfully. You can also verify this by setting a gas limit of 50,000 gas for an ETH transfer transaction and observe how much gas is actually used.

**Reply by EngrPips** [reply]
well said, sir.
